proxy.php

```diff
@@ -407,18 +413,27 @@ function startsWith($string, $with) {
     } elseif ($parsingHeaders) {
         $rest .= $chunk;
         debug_log("build headers to: $rest\n");
-        if (parseLastHeader($rest)) {
+        if (parseLastHeader($rest, $contentLength)) {
             $parsingHeaders = false;
 
             $extOut = fopen("php://output", "w") or errorExit("fundamental error opening PHP output");
             fwrite($extOut, $rest);
+            $contentWritten += strlen($rest);
             $rest = '';
             debug_log("passed last headers");
         }
     } else {
         fwrite($extOut, $chunk);
-        debug_log("proxy : " . strlen($chunk) . " bytes \n");
+        $contentWritten += strlen($chunk);
+        debug_log("proxy : " . strlen($chunk) . " bytes");
     }
+
+    if ($contentLength != -1 && $contentWritten == $contentLength)
+    {
+        debug_log("reached ContentLength of $contentLength bytes");
+        break;
```

Comment:
the new loop-exit using content-length, instead of just waiting for read-eos or so .. good.